src/TwoOutOfThreeColumnVisitor.cpp

```diff
@@ -0,0 +1,148 @@
+                                if (position.first!=-1) {
+                                        if(position.first == 0) {
+                                                if(position.second == 0) {
+                                                        *(regions[0].leftColumn().getFirst()) = value;
+                                                } else if(position.second == 1) {
+                                                        *(regions[0].leftColumn().getSecond()) = value;
+                                                } else if(position.second == 2) {
+                                                        *(regions[0].leftColumn().getThird()) = value;
+                                                }
+                                        } else if(position.first == 1) {
+                                                if(position.second == 0) {
+                                                        *(regions[0].middleColumn().getFirst()) = value;
+                                                } else if(position.second == 1) {
+                                                        *(regions[0].middleColumn().getSecond()) = value;
+                                                } else if(position.second == 2) {
+                                                        *(regions[0].middleColumn().getThird()) = value;
+                                                }
+                                        } else if(position.first == 2) {
+                                                if(position.second == 0) {
+                                                        *(regions[0].rightColumn().getFirst()) = value;
+                                                } else if(position.second == 1) {
+                                                        *(regions[0].rightColumn().getSecond()) = value;
+                                                } else if(position.second == 2) {
+                                                        *(regions[0].rightColumn().getThird()) = value;
+                                                }
+                                        }
+                                        ++changed;
+                                }
+                        }
+                } 
+
+
+                
+        }
+        return changed;
+}
+
+pair<int,int> findOnlyOneNegativeInCols(std::vector<std::vector<bool>> possibles) {
+        pair<int,int> position(-1,-1);
+
+        int totalNegative = 0;
+        for(int i=0; i<3; ++i) {
+                for(int j=0; j<3; ++j) {
+                        if(possibles[i][j] == false) {
+                                totalNegative += 1;
+                                position = pair<int,int>(i,j);
+                        }
+                }
+        }
+
+        if(totalNegative == 1) {
+                return position;
+        }
+
+        return pair<int,int>(-1,-1);
```

Comment:
Same comment as before, should use exception instead of checking the return value (Let's not make this a debate,but that's the "style" of the assignment.